### PR TITLE
perf(solution): remove added files when failure

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

File: packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts [hunk 1]
```diff
@@ -1098,7 +1098,7 @@ export async function createAuthFiles(
         await unzip(sampleZip, tabFolder);
       }
     } catch (error) {
-      // TODO: remove added code
+      await fs.remove(tabFolder);
```
Comment (SLdragon) [inline packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts:1101]: check whether folder exist before remove

Comment (KennethBWSong) [reply packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts:1101]: The folder is created above, it should exist
image

**Review by SLdragon — DISMISSED** [review]

File: packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts (hunk 1 above)
Comment (1yefuwang1) [reply packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts:1101]: Actually, it's possible that OS fails to create the folder and throw an error. So, the folder doesn't necessarily exist here.

Comment (KennethBWSong) [reply packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts]: @1yefuwang1 Error handling here improved. Thx!

**Review by 1yefuwang1 — APPROVED** [review]

**Pull request merged**